Exiled.Loader/Config.cs

```diff
@@ -33,6 +33,12 @@ public sealed class Config : IConfig
         [Description("The working environment type (Development, Testing, Production, Ptb)")]
         public EnvironmentType Environment { get; set; } = EnvironmentType.Production;
 
+        /// <summary>
+        /// Gets or sets a value indicating whether gets or sets the whether to show debug during production.
+        /// </summary>
+        [Description("Whether to allow debugs during production")]
+        public bool DebugProduction { get; set; } = false;
```

Comment:
What in the fuck is this even for? We literally already have config values for showing debugs, the only debugs that aren't shown in the Release build are the #if DEBUG #endif compiler flags, which values like this cannot change since the code just isn't included if that flag isn't met during compilation.

Reply:
Also, to your point about preprocessor directives, yeah that's fair - that is more efficient than a new boolean flag check on log.debug; however, there were examples of that so I went down that route. Agreed

Reply:
Though, I do wonder if it is simpler to just add a new Config EnvironmentType where

EnvironmentType.ProductionDebug where it would reduce this extra bool to just a different loadup config. This would then allow ShouldDebugBeShownProduction to disappear, and just add a flag to ShouldDebugBeShown to check if it is ProductionDebug

Reply:
Adding a new EnvironmentType is also acceptable.